File: js/xdr_generated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Abi = Abi;
exports.FunctionSignature = FunctionSignature;
exports.Parameter = Parameter;
exports.FunctionType = FunctionType;
exports.Account = Account;
exports.AuthorizedAccount = AuthorizedAccount;
exports.RequestType = RequestType;
exports.ResponseType = ResponseType;
exports.Request = Request;
exports.Response = Response;
exports.Block = Block;
exports.BlockHeader = BlockHeader;
exports.BlockHeight = BlockHeight;
exports.BlockStatus = BlockStatus;
exports.Signature = Signature;
exports.ID = ID;
exports.Hash = Hash;
exports.Argument = Argument;
exports.StatusInfo = StatusInfo;
exports.ChannelConfig = ChannelConfig;
exports.Receipt = Receipt;
exports.ReceiptStatus = ReceiptStatus;
exports.Call = Call;
exports.Contract = Contract;
exports.Authorization = Authorization;
exports.Action = Action;
exports.Transaction = Transaction;
exports.AccountUpdateType = AccountUpdateType;
exports.CategoryType = CategoryType;
exports.AccountUpdate = AccountUpdate;
exports.Category = Category;

var _xdrJsSerialize = require("xdr-js-serialize");

var _xdrJsSerialize2 = _interopRequireDefault(_xdrJsSerialize);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function Abi() {
    return new _xdrJsSerialize2.default.Struct(["version", "functions"], [new _xdrJsSerialize2.default.Str('', 0), new _xdrJsSerialize2.default.VarArray(2147483647, FunctionSignature)]);
}
function FunctionSignature() {
    return new _xdrJsSerialize2.default.Struct(["functionType", "functionName", "parameters", "returns"], [FunctionType(), new _xdrJsSerialize2.default.Str('', 2147483647), new _xdrJsSerialize2.default.VarArray(2147483647, Parameter), new _xdrJsSerialize2.default.VarArray(2147483647, Parameter)]);
}
function Parameter() {
    return new _xdrJsSerialize2.default.Struct(["parameterName", "parameterType"], [new _xdrJsSerialize2.default.Str('', 2147483647), new _xdrJsSerialize2.default.Str('', 2147483647)]);
}

// End struct section

// Start enum section

function FunctionType() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "READ",
        2: "WRITE"

    });
}

// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function Account() {
    return new _xdrJsSerialize2.default.Struct(["alias", "transactionCount", "authorizedAccounts"], [new _xdrJsSerialize2.default.Str('', 32), new _xdrJsSerialize2.default.UHyper(), new _xdrJsSerialize2.default.VarArray(32, AuthorizedAccount)]);
}
function AuthorizedAccount() {
    return new _xdrJsSerialize2.default.Struct(["key", "alias"], [ID(), new _xdrJsSerialize2.default.Str('', 32)]);
}

// End struct section

// Start enum section


// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section

// End struct section

// Start enum section

function RequestType() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "TRANSACTION"

    });
}

function ResponseType() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "TRANSACTIONID",
        2: "TRANSACTION",
        3: "TRANSACTIONLIST",
        4: "RECEIPT",
        5: "RECEIPTLIST",
        6: "BLOCK",
        7: "BLOCKLIST",
        8: "BLOCKHEADER",
        9: "BLOCKHEADERLIST",
        10: "CHANNEL",
        11: "CHANNELLIST",
        12: "ACCOUNT",
        13: "HEIGHT",
        14: "ABI"

    });
}

// End enum section

// Start union section


function Request() {
    return new _xdrJsSerialize2.default.Union(RequestType(), {

        "UNKNOWN": () => {
            return new _xdrJsSerialize2.default.Void();
        },

        "TRANSACTION": () => {
            return Transaction();
        }

    });
}

function Response() {
    return new _xdrJsSerialize2.default.Union(ResponseType(), {

        "UNKNOWN": () => {
            return new _xdrJsSerialize2.default.Void();
        },

        "TRANSACTIONID": () => {
            return ID();
        },

        "TRANSACTION": () => {
            return Transaction();
        },

        "TRANSACTIONLIST": () => {
            return new _xdrJsSerialize2.default.VarArray(2147483647, Transaction);
        },

        "RECEIPT": () => {
            return Receipt();
        },

        "RECEIPTLIST": () => {
            return new _xdrJsSerialize2.default.VarArray(2147483647, Receipt);
        },

        "BLOCK": () => {
            return Block();
        },

        "BLOCKLIST": () => {
            return new _xdrJsSerialize2.default.VarArray(2147483647, Block);
        },

        "BLOCKHEADER": () => {
            return BlockHeader();
        },

        "BLOCKHEADERLIST": () => {
            return new _xdrJsSerialize2.default.VarArray(2147483647, BlockHeader);
        },

        "CHANNEL": () => {
            return ChannelConfig();
        },

        "CHANNELLIST": () => {
            return new _xdrJsSerialize2.default.VarArray(2147483647, ChannelConfig);
        },

        "ACCOUNT": () => {
            return Account();
        },

        "HEIGHT": () => {
            return BlockHeight();
        },

        "ABI": () => {
            return Abi();
        }

    });
}

// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function Block() {
    return new _xdrJsSerialize2.default.Struct(["header", "transactions"], [BlockHeader(), new _xdrJsSerialize2.default.VarArray(2147483647, Transaction)]);
}
function BlockHeader() {
    return new _xdrJsSerialize2.default.Struct(["blockHeight", "transactionHeight", "consensusSequenceNumber", "transactionsMerkleRoot", "transactionsReceiptRoot", "stateRoot", "previousHeader", "status"], [new _xdrJsSerialize2.default.UHyper(), new _xdrJsSerialize2.default.UHyper(), new _xdrJsSerialize2.default.UHyper(), Hash(), Hash(), Hash(), Hash(), BlockStatus()]);
}
function BlockHeight() {
    return new _xdrJsSerialize2.default.Struct(["height"], [new _xdrJsSerialize2.default.UHyper()]);
}

// End struct section

// Start enum section

function BlockStatus() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "PENDING",
        2: "FINALIZED"

    });
}

// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section

function Signature() {
    return new _xdrJsSerialize2.default.FixedOpaque(64);
}

function ID() {
    return new _xdrJsSerialize2.default.FixedOpaque(32);
}

function Hash() {
    return new _xdrJsSerialize2.default.FixedOpaque(32);
}

function Argument() {
    return new _xdrJsSerialize2.default.Str('', 2147483647);
}

function StatusInfo() {
    return new _xdrJsSerialize2.default.Str('', 256);
}
// End typedef section

// Start struct section

// End struct section

// Start enum section


// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function ChannelConfig() {
    return new _xdrJsSerialize2.default.Struct(["owner", "admins"], [ID(), new _xdrJsSerialize2.default.VarArray(32, ID)]);
}

// End struct section

// Start enum section


// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function Receipt() {
    return new _xdrJsSerialize2.default.Struct(["status", "stateRoot", "result", "statusInfo"], [ReceiptStatus(), Hash(), new _xdrJsSerialize2.default.Str('', 2147483647), StatusInfo()]);
}

// End struct section

// Start enum section

function ReceiptStatus() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "FAILURE",
        2: "SUCCESS"

    });
}

// End enum section

// Start union section


// End union section

// End namespace mazzaroth
// Namespace start mazzaroth

// Start typedef section
// End typedef section

// Start struct section
function Call() {
    return new _xdrJsSerialize2.default.Struct(["function", "arguments"], [new _xdrJsSerialize2.default.Str('', 256), new _xdrJsSerialize2.default.VarArray(2147483647, Argument)]);
}
function Contract() {
    return new _xdrJsSerialize2.default.Struct(["contractBytes", "abi", "contractHash", "version"], [new _xdrJsSerialize2.default.VarOpaque(2147483647), Abi(), Hash(), new _xdrJsSerialize2.default.Str('', 100)]);
}
function Authorization() {
    return new _xdrJsSerialize2.default.Struct(["account", "authorize"], [AuthorizedAccount(), new _xdrJsSerialize2.default.Bool()]);
}
function Action() {
    return new _xdrJsSerialize2.default.Struct(["channelID", "nonce", "blockExpirationNumber", "category"], [ID(), new _xdrJsSerialize2.default.UHyper(), new _xdrJsSerialize2.default.UHyper(), Category()]);
}
function Transaction() {
    return new _xdrJsSerialize2.default.Struct(["signature", "sender", "signer", "action"], [Signature(), ID(), ID(), Action()]);
}

// End struct section

// Start enum section

function AccountUpdateType() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "ALIAS",
        2: "AUTHORIZATION"

    });
}

function CategoryType() {
    return new _xdrJsSerialize2.default.Enum({
        0: "UNKNOWN",
        1: "CALL",
        2: "CONTRACT",
        3: "CONFIG",
        4: "ACCOUNT"

    });
}

// End enum section

// Start union section


function AccountUpdate() {
    return new _xdrJsSerialize2.default.Union(AccountUpdateType(), {

        "UNKNOWN": () => {
            return new _xdrJsSerialize2.default.Void();
        },

        "ALIAS": () => {
            return new _xdrJsSerialize2.default.Str('', 0);
        },

        "AUTHORIZATION": () => {
            return Authorization();
        }

    });
}

function Category() {
    return new _xdrJsSerialize2.default.Union(CategoryType(), {

        "UNKNOWN": () => {
            return new _xdrJsSerialize2.default.Void();
        },

        "CALL": () => {
            return Call();
        },

        "CONTRACT": () => {
            return Contract();
        },

        "CONFIG": () => {
            return ChannelConfig();
        },

        "ACCOUNT": () => {
            return AccountUpdate();
        }

    });
}

// End union section

// End namespace mazzaroth

File: rust/xdr_generated.rs
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#[macro_use]
extern crate xdr_rs_serialize_derive;
#[allow(unused_imports)]
use std::io::Write;
#[allow(unused_imports)]
use xdr_rs_serialize::de::{
    read_fixed_array, read_fixed_array_json, read_fixed_opaque, read_fixed_opaque_json,
    read_var_array, read_var_array_json, read_var_opaque, read_var_opaque_json, read_var_string,
    read_var_string_json, XDRIn,
};
use xdr_rs_serialize::error::Error;
#[allow(unused_imports)]
use xdr_rs_serialize::ser::{
    write_fixed_array, write_fixed_array_json, write_fixed_opaque, write_fixed_opaque_json,
    write_var_array, write_var_array_json, write_var_opaque, write_var_opaque_json,
    write_var_string, write_var_string_json, XDROut,
};

extern crate json;

// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Abi {
    pub version: String,

    #[array(var = 2147483647)]
    pub functions: Vec<FunctionSignature>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct FunctionSignature {
    pub functionType: FunctionType,

    #[array(var = 2147483647)]
    pub functionName: String,

    #[array(var = 2147483647)]
    pub parameters: Vec<Parameter>,

    #[array(var = 2147483647)]
    pub returns: Vec<Parameter>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Parameter {
    #[array(var = 2147483647)]
    pub parameterName: String,

    #[array(var = 2147483647)]
    pub parameterType: String,
}

// End struct section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum FunctionType {
    UNKNOWN = 0,
    READ = 1,
    WRITE = 2,
}

impl Default for FunctionType {
    fn default() -> Self {
        FunctionType::UNKNOWN
    }
}
// Start union section

// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Account {
    #[array(var = 32)]
    pub alias: String,

    pub transactionCount: u64,

    #[array(var = 32)]
    pub authorizedAccounts: Vec<AuthorizedAccount>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct AuthorizedAccount {
    pub key: ID,

    #[array(var = 32)]
    pub alias: String,
}

// End struct section

// Start union section

// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

// End struct section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum RequestType {
    UNKNOWN = 0,
    TRANSACTION = 1,
}

impl Default for RequestType {
    fn default() -> Self {
        RequestType::UNKNOWN
    }
}

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum ResponseType {
    UNKNOWN = 0,
    TRANSACTIONID = 1,
    TRANSACTION = 2,
    TRANSACTIONLIST = 3,
    RECEIPT = 4,
    RECEIPTLIST = 5,
    BLOCK = 6,
    BLOCKLIST = 7,
    BLOCKHEADER = 8,
    BLOCKHEADERLIST = 9,
    CONFIG = 10,
    ACCOUNT = 11,
    HEIGHT = 12,
    ABI = 13,
}

impl Default for ResponseType {
    fn default() -> Self {
        ResponseType::UNKNOWN
    }
}
// Start union section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum Request {
    UNKNOWN(()),

    TRANSACTION(Transaction),
}

impl Default for Request {
    fn default() -> Self {
        Request::UNKNOWN(())
    }
}

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum Response {
    UNKNOWN(()),

    TRANSACTIONID(ID),

    TRANSACTION(Transaction),

    #[array(var = 2147483647)]
    TRANSACTIONLIST(Vec<Transaction>),

    RECEIPT(Receipt),

    #[array(var = 2147483647)]
    RECEIPTLIST(Vec<Receipt>),

    BLOCK(Block),

    #[array(var = 2147483647)]
    BLOCKLIST(Vec<Block>),

    BLOCKHEADER(BlockHeader),

    #[array(var = 2147483647)]
    BLOCKHEADERLIST(Vec<BlockHeader>),

    CONFIG(Config),

    ACCOUNT(Account),

    HEIGHT(BlockHeight),

    ABI(Abi),
}

impl Default for Response {
    fn default() -> Self {
        Response::UNKNOWN(())
    }
}
// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Block {
    pub header: BlockHeader,

    #[array(var = 2147483647)]
    pub transactions: Vec<Transaction>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct BlockHeader {
    pub blockHeight: u64,

    pub transactionHeight: u64,

    pub consensusSequenceNumber: u64,

    pub transactionsMerkleRoot: Hash,

    pub transactionsReceiptRoot: Hash,

    pub stateRoot: Hash,

    pub previousHeader: Hash,

    pub status: Status,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct BlockHeight {
    pub height: u64,
}

// End struct section

// Start union section

// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Signature {
    #[array(fixed = 64)]
    pub t: Vec<u8>,
}
#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct ID {
    #[array(fixed = 32)]
    pub t: Vec<u8>,
}
#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Hash {
    #[array(fixed = 32)]
    pub t: Vec<u8>,
}
#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Argument {
    #[array(var = 2147483647)]
    pub t: String,
}
#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct StatusInfo {
    #[array(var = 256)]
    pub t: String,
}

// End typedef section

// Start struct section

// End struct section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum Status {
    UNKNOWN = 0,
    SUCCESS = 1,
    FAILURE = 2,
    PENDING = 3,
    FINALIZED = 4,
}

impl Default for Status {
    fn default() -> Self {
        Status::UNKNOWN
    }
}
// Start union section

// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Receipt {
    pub status: Status,

    pub stateRoot: Hash,

    #[array(var = 2147483647)]
    pub result: String,

    pub statusInfo: StatusInfo,
}

// End struct section

// Start union section

// End union section

// Namspace end mazzaroth
// Namspace start mazzaroth

// Start typedef section

// End typedef section

// Start struct section

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Call {
    #[array(var = 256)]
    pub function: String,

    #[array(var = 2147483647)]
    pub arguments: Vec<Argument>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Config {
    pub owner: ID,

    #[array(var = 32)]
    pub admins: Vec<ID>,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Contract {
    #[array(var = 2147483647)]
    pub contractBytes: Vec<u8>,

    pub abi: Abi,

    pub contractHash: Hash,

    #[array(var = 100)]
    pub version: String,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Authorization {
    pub account: AuthorizedAccount,

    pub authorize: bool,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Data {
    pub channelID: ID,

    pub nonce: u64,

    pub blockExpirationNumber: u64,

    pub category: Category,
}

#[derive(PartialEq, Clone, Default, Debug, XDROut, XDRIn)]
pub struct Transaction {
    pub sender: ID,

    pub signer: ID,

    pub signature: Signature,

    pub data: Data,
}

// End struct section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum AccountUpdateType {
    UNKNOWN = 0,
    ALIAS = 1,
    AUTHORIZATION = 2,
}

impl Default for AccountUpdateType {
    fn default() -> Self {
        AccountUpdateType::UNKNOWN
    }
}

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum CategoryType {
    UNKNOWN = 0,
    CALL = 1,
    CONTRACT = 2,
    CONFIG = 3,
    ACCOUNT = 4,
}

impl Default for CategoryType {
    fn default() -> Self {
        CategoryType::UNKNOWN
    }
}
// Start union section

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum AccountUpdate {
    UNKNOWN(()),

    ALIAS(String),

    AUTHORIZATION(Authorization),
}

impl Default for AccountUpdate {
    fn default() -> Self {
        AccountUpdate::UNKNOWN(())
    }
}

#[derive(PartialEq, Clone, Debug, XDROut, XDRIn)]
pub enum Category {
    UNKNOWN(()),

    CALL(Call),

    CONTRACT(Contract),

    CONFIG(Config),

    ACCOUNT(AccountUpdate),
}

impl Default for Category {
    fn default() -> Self {
        Category::UNKNOWN(())
    }
}
// End union section

// Namspace end mazzaroth
